tests/dragonfly/memory_test.py

```diff
@@ -51,9 +52,10 @@ async def test_rss_used_mem_gap(df_server: DflyInstance, type, keys, val_size, e
     # It could be the case that the machine is configured to use swap if this assertion fails
     assert delta > 0
     assert delta < max_unaccounted
-    delta = info["used_memory_rss"] - info["object_used_memory"]
-    # TODO investigate why it fails on string
-    if type == "JSON" or type == "STREAM":
+
+    if type != "LIST":
+        delta = info["used_memory_rss"] - info["object_used_memory"]
+        max_unaccounted *= 1.1  # Some more memory is needed for dash table, keys, etc
```

Comment:
so used_memory counts the dash table and the key memory while object_used_memory does not there for the delta is bigger
Not sure why we compare in this test the rss to object_used_memory in this case
does is makes sense to compare it to val_size*elements_count ?

Reply:
That's a very good idea. Hopefully the overhead for that could be low and relatively equal for all types.

Reply:
Unfortunately, it looks like your proposal (which I liked a lot!) won't fly..

These are the values we use to try to generate 3gb of RSS. All is pretty optimized except for string, I think.

  | keys | elements | element sz | total
-- | -- | -- | -- | --
"JSON" | 250,000 | 100 | 75 | 1,875,000,000
"SET" | 250,000 | 100 | 110 | 2,750,000,000
"HASH" | 250,000 | 100 | 100 | 2,500,000,000
"ZSET" | 250,000 | 100 | 100 | 2,500,000,000
"LIST" | 250,000 | 100 | 125 | 3,125,000,000
"STRING" | 250,000 | 20,000 | 1 | 5,000,000,000
"STREAM" | 250,000 | 100 | 120 | 3,000,000,000

So the overhead is quite different, and I don't know if it makes much sense to compare that to anything. I mean, sure, JSON has a huge overhead.. what's there to test?

After thinking about this some more, perhaps we should just remove that last part of the test :shrug:

Reply:
I believe that by this test we found that we dont account correctly object memory for steams.
Therefore I suggest to ease the check in this test and to check that object_used_memory > keys*elements*elements_size and object_used_memory < used_memory
This way we will still have some check to catch extreme cases where we somehow drop counting